src/python/foglamp/configuration_manager.py

```diff
@@ -147,22 +154,32 @@ def _run_callbacks(category_name):
 
 
 async def _create_new_category(category_name, category_val, category_description):
-    async with aiopg.sa.create_engine(_connection_string) as engine:
-        async with engine.acquire() as conn:
-            await conn.execute(_configuration_tbl.insert().values(key=category_name, value=category_val,
-                                                                  description=category_description))
+    payload = PayloadBuilder().INSERT(key=category_name, description=category_description, value=category_val).payload()
+    _storage = connect.get_storage()
+    _storage.insert_into_tbl("configuration", payload)
 
 
 async def _read_all_category_names():
-    async with aiopg.sa.create_engine(_connection_string) as engine:
-        async with engine.acquire() as conn:
-            category_info = []
-            async for row in conn.execute(_configuration_tbl.select()):
-                category_info.append((row.key, row.description))
-            return category_info
+    # SELECT configuration.key, configuration.description, configuration.value, configuration.ts FROM configuration
+    payload = PayloadBuilder().SELECT(("key", "description", "value", "ts")).payload()
+    _storage = connect.get_storage()
+    results = _storage.query_tbl_with_payload('configuration', payload)
+
+    category_info = []
+    for row in results['rows']:
+        category_info.append((row['key'], row['description']))
+    return category_info
 
 
 async def _read_category_val(category_name):
+    # SELECT configuration.key, configuration.description, configuration.value,
+    # configuration.ts FROM configuration WHERE configuration.key = :key_1
+    # payload = PayloadBuilder().SELECT(("value"))\
+    #     .WHERE(["key", "=", category_name]).payload()
+    # _storage = connect.get_storage()
+    # results = _storage.query_tbl_with_payload('configuration', payload)
+    # for row in results['rows']:
+    #     return row['value']
     async with aiopg.sa.create_engine(_connection_string) as engine:
         async with engine.acquire() as conn:
             async for row in conn.execute(
```

Comment:
Kindly advise!

Reply:
The code in the comment looks right to be, I would added a return item to limit what is returned, but this is not a requirement. E.g. 
{
	"where" : {
			"column" : "key",
 			"condition" : "=",
			"value" : "SCHEDULER"
		},
	"return" : [ "value" ]
}

Where the key is SCHEUDLER